src/OSInfo.cpp

```diff
@@ -34,7 +34,10 @@ static QString regString(wchar_t *string, int size) {
 	// that the NUL is not included in the returned
 	// string.
 	const size_t adjustedSize = wcsnlen(string, static_cast<size_t>(size));
-	return QString::fromWCharArray(string, adjustedSize);
+	if (adjustedSize > INT_MAX) {
+		return QString();
```

Comment:
This is unreachable. The return value of wcsnlen is <= size which is itself an int.